code/modules/mob/dead/observer/observer.dm

```diff
@@ -446,6 +447,29 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 			else
 				A << "This mob is not located in the game world."
 
+/mob/dead/observer/verb/change_view_range()
+	set category = "Ghost"
+	set name = "Modify View Range"
+	set desc = "Change your view range."
+
+	var/max_view = client.prefs.unlock_content ? GHOST_MAX_VIEW_RANGE_MEMBER : GHOST_MAX_VIEW_RANGE_DEFAULT
+	if(client.view == world.view)
+		var/list/views = list()
+		for(var/i in 1 to max_view)
+			views |= i
+		var/new_view = input("Choose your new view", "Modify view range", 7) as null|anything in views
+		if(new_view)
+			client.view = Clamp(new_view, 1, max_view)
+	else
+		client.view = world.view
+
+/mob/dead/observer/verb/add_view_range(input as text)
+	set name = "Add View Range"
+
+	var/max_view = client.prefs.unlock_content ? 14 : 10
+	if(input)
+		client.view = Clamp(client.view + text2num(input), 1, max_view)
+
```

Comment:
This one is called from the skin.dmf, I was not sure if you could do as num and it would convert automatically.